libs/httpproxy/httpproxy_test.go

```diff
@@ -0,0 +1,144 @@
+package httpproxy
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/databricks/cli/libs/testserver"
+	"github.com/stretchr/testify/require"
+
+	"github.com/gorilla/websocket"
+)
+
+const (
+	PROXY_PORT = 8081
+)
+
+var (
+	PROXY_ADDR = fmt.Sprintf("localhost:%d", PROXY_PORT)
+	PROXY_URL  = "http://" + PROXY_ADDR
+)
+
+func sendTestRequest(t *testing.T, path string) (int, []byte) {
+	req, err := http.NewRequest("GET", PROXY_URL+path, bytes.NewBufferString("{'test': 'value'}"))
+	require.NoError(t, err)
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := http.DefaultClient.Do(req)
+	require.NoError(t, err)
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	require.NoError(t, err)
+
+	return resp.StatusCode, body
+}
+
+func startProxy(t *testing.T, serverAddr string) *Proxy {
+	proxy, err := NewProxy("http://" + serverAddr)
+	require.NoError(t, err)
+
+	go func() {
+		_ = proxy.Start(PROXY_ADDR)
+	}()
+
+	// Wait for the server to start
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Is it waiting for the proxy to start? Can we block until it starts? Would we need to sleep when the library is actually used? 

In that case, blocking would be better.

Reply:
If you split the calls to `Listen` and `Serve` in the implementation, then you don't need to wait.